From 4bf612e4cd941193a774ed37de98a394682efa8d Mon Sep 17 00:00:00 2001
From: Luc Schrijvers <begasus@gmail.com>
Date: Sat, 14 Jun 2025 12:11:07 +0200
Subject: Revert: Determine direction from language

Fixes the need for changes in harfbuzz recipe

diff --git a/gtk/gtkmain.c b/gtk/gtkmain.c
index 86f7307..18e2892 100644
--- a/gtk/gtkmain.c
+++ b/gtk/gtkmain.c
@@ -48,8 +48,6 @@
 #undef STRICT
 #endif
 
-#include <hb-glib.h>
-
 #include <glib/gi18n-lib.h>
 
 #include "gtkdebug.h"
@@ -773,37 +771,18 @@ gtk_is_initialized (void)
 GtkTextDirection
 gtk_get_locale_direction (void)
 {
-  PangoLanguage *language;
-  const PangoScript *scripts;
-  int n_scripts;
-
-  language = gtk_get_default_language ();
-  scripts = pango_language_get_scripts (language, &n_scripts);
-
-  if (n_scripts > 0)
-    {
-      for (int i = 0; i < n_scripts; i++)
-        {
-          hb_script_t script;
-
-          script = hb_glib_script_to_script ((GUnicodeScript) scripts[i]);
-
-          switch (hb_script_get_horizontal_direction (script))
-            {
-            case HB_DIRECTION_LTR:
-              return GTK_TEXT_DIR_LTR;
-            case HB_DIRECTION_RTL:
-              return GTK_TEXT_DIR_RTL;
-            case HB_DIRECTION_TTB:
-            case HB_DIRECTION_BTT:
-            case HB_DIRECTION_INVALID:
-            default:
-              break;
-            }
-        }
-    }
-
-  return GTK_TEXT_DIR_LTR;
+  /* Translate to default:RTL if you want your widgets
+   * to be RTL, otherwise translate to default:LTR.
+   * Do *not* translate it to "predefinito:LTR", if it
+   * it isn't default:LTR or default:RTL it will not work
+   */
+  char             *e   = _("default:LTR");
+  GtkTextDirection  dir = GTK_TEXT_DIR_LTR;
+  if (g_strcmp0 (e, "default:RTL") == 0)
+    dir = GTK_TEXT_DIR_RTL;
+  else if (g_strcmp0 (e, "default:LTR") != 0)
+    g_warning ("Whoever translated default:LTR did so wrongly. Defaulting to LTR.");
+ return dir;
 }
 
 /**
-- 
2.48.1

